File: Web/TSSampleProject/src/PostgreClientExample.ts
import { getLogger } from './log/log.js';
import { closePool } from "./postgre/db.js";
import postgreBaseAsync from "./postgre/PostgreBase.js";
import postgreOrmAsync from "./postgre/PostgreOrm.js";

// 日志
const log = getLogger();

// 主函数
async function main(): Promise<void> {
	try {
		log.info("开始运行PostgreBase");
		// 调用Postgre基础实例
		await postgreBaseAsync();
		await closePool();
		log.info("===================");
		log.info("开始运行PostgreOrm");
		// 调用PostgreORM实例
		await postgreOrmAsync();
	} catch (err) {
		throw err;
	} finally {
	}
}

// 调用 main 函数
main()
	.then(async () => {
		log.info("PostgreClient运行成功");
	})
	.catch((err) => {
		log.error(err, "PostgreClient发生错误");
	}).finally(async () => {
		log.flush();
	});

File: Rust/rust_sample/src/postgre_client_example.rs
use std::error::Error;
use tracing as log;

// 导入utils模块（utils.rs）
mod utils;
// 导入postgre模块（postgre.rs）
mod postgre;

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // 初始化日志
    utils::log::init_log_async().await?;
    // Postgre的基础示例
    let ret = postgre::postgre_base::postgre_base_async().await;
    match ret {
        Ok(()) => log::info!("PostgreBase示例运行结束"),
        Err(e) => {
            log::error!("PostgreBase示例运行失败: {}", e)
        }
    }
    log::info!("=====================================");
    // Postgre的sea-orm示例
    let ret = postgre::postgre_sea_orm::postgre_sea_orm_async().await;
    match ret {
        Ok(()) => log::info!("PostgreSeaOrm示例运行结束"),
        Err(e) => {
            log::error!("PostgreSeaOrm示例运行失败: {}", e)
        }
    }

    Ok(())
}
